docs/entities/serviceclaim.md

```diff
@@ -0,0 +1,48 @@
+# ServiceClaim
+
+A Service Claim represents a claim for service.
+
+## Specification
+
+The definition of a ServiceClaim can be obtained directly from our [ServiceClaim
+CRD](../../config/crd/bases/primaza.io_serviceclaims.yaml). The specification
+contains two required properties:
+
+- ServiceClassIdentity: A set of key/value pairs that identify the service
+  class. Examples of service class identity keys include type of service, and
+  provider of service. This property is required.
+- ServiceEndpointDefinitionKeys: An array of keys that is required for
+  connectivity. The values corresponding to each of these keys will be extracted
+  from the service. This property is required.
+
+There are other three optional fields. The EnvironmentTag and
+ApplicationClusterContext are mutually exclusive:
+
+- Application: Fields indentifies application resources through kind, apiVersion and label selector. 
+- EnvironmentTag: A string representing one of the environment.
+- ApplicationClusterContext:
+
+## Status
+
+The Status of the ServiceClaim is also defined under the [ServiceClaim
+CRD](../../config/crd/bases/primaza.io_serviceclaims.yaml). It contains only one
+required property to track state. The state could be either `Pending` or
+`Resolved`. If the state is `Resolved`, there should be Secret and Service
+Binding resources created
+
+There is a `claimID` fields with a unique ID for the claim.  And there is another field, `registeredService` that points the service.
+
+## Use Cases
+
+### Creation
+
+When a Service Claim is created, Primaza should find a service based on the given input and create Secret and Service Binding resources.  Then it should the the state accordingly.
```

Comment:
I think you should expand this on what find a service means. Not to much details but the fact that we are going to try to match SCI and SED Keys with a Registered service.